lock-impl/src/main/java/com/palantir/lock/impl/LockServiceImpl.java

```diff
@@ -142,6 +140,10 @@ public String apply(HeldLocksToken from) {
             this.locks = locks;
         }
 
+        public List<LockDescriptor> getLockDescriptors() {
+            return locks.stream().map(ClientAwareReadWriteLock::getDescriptor).collect(Collectors.toList());
```

Comment:
Likely not a problem because this isn't on a hot code path, but streams for stuff like this have been shown to be significantly slower - https://jaxenter.com/java-performance-tutorial-how-fast-are-the-java-8-streams-118830.html